test/view/broadcast/broadcast_round_screen_test.dart

```diff
@@ -0,0 +1,253 @@
+import 'package:lichess_mobile/src/widgets/platform.dart';
+import 'package:network_image_mock/network_image_mock.dart';
+
+import '../../network/fake_websocket_channel.dart';
+import '../../test_helpers.dart';
+import '../../test_provider_scope.dart';
+
+final client = MockClient((request) {
+  if (request.url.path == '/api/broadcast/RAIoMC7L') {
+    return mockResponse(
+      _tournamentResponse,
+      200,
+      headers: {'content-type': 'application/json; charset=utf-8'},
+    );
+  }
+  if (request.url.path == '/api/broadcast/-/-/Gv1305pb') {
+    return mockResponse(
+      _roundResponse,
+      200,
+      headers: {'content-type': 'application/json; charset=utf-8'},
+    );
+  }
+  return mockResponse('', 404);
+});
+
+void main() {
+  group('Broadcast round screen', () {
+    testWidgets('Displays broadcast round screen', variant: kPlatformVariant, (tester) async {
+      final fakeSocket = FakeWebSocketChannel();
+      final app = await makeTestProviderScopeApp(
+        tester,
+        home: BroadcastRoundScreen(broadcast: _broadcast),
+        overrides: [
+          lichessClientProvider.overrideWith((ref) => LichessClient(client, ref)),
+          webSocketChannelFactoryProvider.overrideWith((ref) {
+            return FakeWebSocketChannelFactory((_) => fakeSocket);
+          }),
+        ],
+      );
+
+      await tester.pumpWidget(app);
+
+      expect(find.byType(CircularProgressIndicator), findsOneWidget);
+
+      // Load the tournament
+      await tester.pump();
+
+      expect(find.byType(CircularProgressIndicator), findsOneWidget);
+
+      // Load the round
+      await tester.pump();
+
+      expect(find.byType(BroadcastBoardsTab), findsOneWidget);
+      expect(find.byType(BoardThumbnail), findsNWidgets(4));
+      expect(find.byType(BroadcastPlayerWidget), findsNWidgets(8));
+      expect(find.text('Yilmaz, Mustafa'), findsOneWidget);
+      expect(find.text('Gokerkan, Cem Kaan'), findsOneWidget);
+    });
+
+    testWidgets(
+      'Swipe to overview tab on Android',
+      variant: TargetPlatformVariant.only(TargetPlatform.android),
+      (tester) async {
+        mockNetworkImagesFor(() async {
+          final fakeSocket = FakeWebSocketChannel();
+          final app = await makeTestProviderScopeApp(
+            tester,
+            home: BroadcastRoundScreen(broadcast: _broadcast),
+            overrides: [
+              lichessClientProvider.overrideWith((ref) => LichessClient(client, ref)),
+              webSocketChannelFactoryProvider.overrideWith((ref) {
+                return FakeWebSocketChannelFactory((_) => fakeSocket);
+              }),
+            ],
+          );
+
+          await tester.pumpWidget(app);
+
+          // Load the tournament
+          await tester.pump();
+
+          // Load the round
+          await tester.pump();
+
+          // Swipe to the overview tab
+          await tester.flingFrom(const Offset(50, 300), const Offset(500, 0), 1000);
```

Comment:
Don't bother to use a gesture when you can use a `tester.tap(find.text('Overview'))`.